steps/jwt/session_test.go

```diff
@@ -0,0 +1,896 @@
+		// 		ctx:          context.Background(),
+		// 		publicKeyPEM: invalidPublicKey,
+		// 	},
+		// },
+		{
+			name:    "Default",
+			wantErr: true,
+			args: args{
+				ctx:          context.Background(),
+				publicKeyPEM: wrongTypeKey,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Session{}
+			if err := s.ConfigurePublicKey(tt.args.ctx, tt.args.publicKeyPEM); (err != nil) != tt.wantErr {
+				t.Errorf("Session.ConfigurePublicKey() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestConfigurePrivateKey(t *testing.T) {
+	type args struct {
+		ctx           context.Context
+		privateKeyPEM string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Not valid pem",
+			wantErr: true,
+			args: args{
+				ctx:           context.Background(),
+				privateKeyPEM: "WRONG",
+			},
+		},
+		{
+			name:    "Valid Private Key",
+			wantErr: false,
+			args: args{
+				ctx:           context.Background(),
+				privateKeyPEM: signPrivateKey,
+			},
+		},
+		// {
+		// 	name:    "Invalid Key",
+		// 	wantErr: true,
+		// 	args: args{
+		// 		ctx:          context.Background(),
+		// 		publicKeyPEM: invalidPublicKey,
+		// 	},
+		// },
+		{
+			name:    "Default",
+			wantErr: true,
+			args: args{
+				ctx:           context.Background(),
+				privateKeyPEM: wrongTypeKey,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Session{}
+			if err := s.ConfigurePrivateKey(tt.args.ctx, tt.args.privateKeyPEM); (err != nil) != tt.wantErr {
+				t.Errorf("Session.ConfigurePrivateKey() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestConfigureJSONPayload(t *testing.T) {
+	paramsInput := make(map[string]interface{})
+	paramsInput["title"] = "foo1"
+	type args struct {
+		ctx   context.Context
+		props map[string]interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Valid payload",
+			args: args{
+				ctx:   context.Background(),
+				props: paramsInput,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Session{}
+			if err := s.ConfigureJSONPayload(tt.args.ctx, tt.args.props); (err != nil) != tt.wantErr {
+				t.Errorf("Session.ConfigureJSONPayload() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestGenerateSignedJWTInContext(t *testing.T) {
+	type args struct {
+		ctx     context.Context
+		ctxtKey string
+	}
+	tests := []struct {
+		name               string
+		args               args
+		payload            []byte
+		signatureAlgorithm jwa.SignatureAlgorithm
+		privateKey         interface{}
+		wantErr            bool
+	}{
+		{
+			name:    "Nil payload",
+			payload: nil,
+			wantErr: true,
+			args: args{
+				ctx: context.Background(),
+			},
+		},
+		{
+			name:               "Empty signature",
+			payload:            []byte("payload"),
```

Comment:
add to constant to avoid duplicities:
bytePayload []byte = []byte("payload")

Reply:
Moved to global var (it could not be a constant)